motionplan/motionPlanner.go

```diff
@@ -511,6 +516,41 @@ func CheckPlan(
 		return err
 	}
 
+	// create a list of segments to iterate through
+	var segments []*ik.Segment
+	if relative {
```

Comment:
hmmm, I do not think `createSegment` is what we want here.
Suppose we our remaining plan is `[[1,2,1000],[0,0,0]]`. Further suppose our current inputs are `[1,2,300]`.
For us to properly interpolate the we need to go from `[1,2,300], [1,2,301], ... ,[1,2,999], [1,2,1000]`

If we use `create segment` we would instead be doing
`[1,2,1], [1,2,2], [1,2,3], ... ,[1,2,999], [1,2,1000]`
this would then return in checking checking something we have already executed.